### PR TITLE
[Chat] Verify the fetch endpoints match

### PR DESCRIPTION
**Pull request by JoshuaLai**
# What
Verify the endpoints match before making the call

# Why
Adding security to the fetch call to not pass the token into a call sourced from a different endpoint. 

# How Tested
Running call with chat scenario all inline images should work as expected 

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/chat-stateful-client/src/ChatContext.ts [hunk 1]
```diff
@@ -48,14 +48,15 @@ export class ChatContext {
...
+    /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */ endpoint?: string
   ) {
     this._logger = createClientLogger('communication-react:chat-context');
     this._emitter = new EventEmitter();
     /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
     if (credential) {
-      this._inlineImageQueue = new ResourceDownloadQueue(this, credential);
-      this._fullsizeImageQueue = new ResourceDownloadQueue(this, credential);
+      this._inlineImageQueue = new ResourceDownloadQueue(this, { credential, endpoint: endpoint ?? '' });
+      this._fullsizeImageQueue = new ResourceDownloadQueue(this, { credential, endpoint: endpoint ?? '' });
```
Comment (jpeng-ms) [inline packages/chat-stateful-client/src/ChatContext.ts:59]: in what case would `endpoint` be undefined?

Comment (JoshuaLai) [reply packages/chat-stateful-client/src/ChatContext.ts:59]: This is a good question. I think the implementation and arguments can be cleaned up a bit, but technically the credential and the endpoint in this case should go hand-in-hand. There is no point in passing the credential without the endpoint, and no point in passing the endpoint without the credential (so maybe they should be put into a one object).

Comment (emlynmac) [reply packages/chat-stateful-client/src/ChatContext.ts:59]: Is this not a question of source compatibility with previous API versions? If this is non-optional, it breaks existing callers of this public API.

Comment (emlynmac) [reply packages/chat-stateful-client/src/ChatContext.ts:59]: The constructor takes an optional endpoint, as it has to be optional to maintain source compatibility.

We can either propagate this optional or define it to be non-optional like this.

File: packages/chat-stateful-client/src/ResourceDownloadQueue.ts [hunk 2]
```diff
@@ -188,7 +194,12 @@ export const fetchImageSource = async (
     clearTimeout(id);
     return response;
   }
-  const accessToken = await credential.getToken();
+
+  const accessToken = await authentication.credential.getToken();
+  if (!src.startsWith(authentication.endpoint)) {
```
Comment (emlynmac) [inline packages/chat-stateful-client/src/ResourceDownloadQueue.ts:199]: This must not be `startsWith`.
We need to check that the root of the domain of the URL is `communication.azure.com`
We must also verify that the protocol is `https`

Comment (emlynmac) [inline packages/chat-stateful-client/src/ResourceDownloadQueue.ts:199]: Use https://developer.mozilla.org/en-US/docs/Web/API/URL/hostname
 to check the host name and protocol type

File: packages/chat-stateful-client/src/ResourceDownloadQueue.test.ts [hunk 3]
```diff
@@ -71,7 +71,7 @@ describe('ResourceDownloadQueue api functions', () => {
     const editedMessage = { ...originalMessage };
     editedMessage.content = { message: 'edited message', attachments: secondAttachments };
 
-    const queue = new ResourceDownloadQueue(context, tokenCredential);
+    const queue = new ResourceDownloadQueue(context, { credential: tokenCredential, endpoint: 'endpoint' });
```
Comment (emlynmac) [inline packages/chat-stateful-client/src/ResourceDownloadQueue.test.ts:74]: Let's not use magic text?

Comment (JoshuaLai) [reply packages/chat-stateful-client/src/ResourceDownloadQueue.test.ts:74]: This is a test file, so we kind of have magic text everywhere. Should we just refactor the whole test file?

Comment (emlynmac) [reply packages/chat-stateful-client/src/ResourceDownloadQueue.test.ts:74]: Extracting this to a const value doesn't seem like much.

**Review by Leah-Xia-Microsoft — APPROVED** [review]

**Review by jpeng-ms — APPROVED** [review]

**Review by emlynmac — APPROVED** [review]

**Pull request merged**